opt_adapt/opt.py

```diff
@@ -632,4 +632,4 @@ def minimise(
 
         # Clean up
         tape.clear_tape()
-    return u_plus
+    return u_plus, mesh
```

Comment:
I think it would be nicer if you could instead store the mesh from each iteration in the `OptimisationProgress` class, like we do with the values of the control, objective function, etc.

To copy a mesh, I think `copied_mesh = Mesh(mesh.coordinates.copy(deepcopy=True))` is sufficient.